index.js

```diff
@@ -1,10 +1,114 @@
 'use strict';
 /* eslint-env node */
-var SassCompilerFactory = require('broccoli-sass-source-maps');
-var path = require('path');
-var VersionChecker = require('ember-cli-version-checker');
-var Funnel = require('broccoli-funnel');
-var mergeTrees = require('broccoli-merge-trees');
+
+const path = require('path');
+const Filter = require('broccoli-persistent-filter');
+const mkdirp = require('mkdirp');
+const VersionChecker = require('ember-cli-version-checker');
+const Funnel = require('broccoli-funnel');
+const mergeTrees = require('broccoli-merge-trees');
+const fs = require('fs');
+const includePathSearcher = require('include-path-searcher');
+
+class SassCompiler extends Filter {
+  constructor(inputTree, inputOutputMap, _options) {
+    let options = _options || {};
+    if (!options || typeof options !== 'object') {
+      options = { persist: true };
+    } else if (typeof options.persist === 'undefined') {
+      options.persist = true;
+    }
+
+    super(inputTree, options);
+
+    this.name = 'sass-compiler';
+    this.options = options;
+    this.inputTree = inputTree;
+    this.inputOutputMap = inputOutputMap;
+    this.lastBuildStart = undefined;
+    this.extensions = ['scss', 'sass'];
+    this.targetExtension = 'css';
+
+    this.renderSassSync = options.implementation.renderSync;
+
+    this.sassOptions = {
+      importer: options.importer,
+      functions: options.functions,
+      indentedSyntax: options.indentedSyntax,
+      omitSourceMapUrl: options.omitSourceMapUrl,
+      outputStyle: options.outputStyle,
+      precision: options.precision,
+      sourceComments: options.sourceComments,
+      sourceMap: options.sourceMap,
+      sourceMapEmbed: options.sourceMapEmbed,
+      sourceMapContents: options.sourceMapContents,
+      sourceMapRoot: options.sourceMapRoot,
+      fiber: options.fiber
+    };
+  }
+
+  baseDir() {
+    return __dirname;
+  }
+
+  processString() {
+    if(this.inputTree._buildStart !== this.lastBuildStart) {
+      this.inputOutputMap.forEach(({ input, output}) => {
+        var destFile = path.join(this.outputPath, output);
+        var sourceMapFile = this.sassOptions.sourceMap;
+
+        if (typeof sourceMapFile !== 'string') {
+          sourceMapFile = destFile + '.map';
+        }
+
+        mkdirp.sync(path.dirname(destFile));
```

Comment:
Speculative `mkdirp.sync` is somewhat expensive, oftentimes we can avoid it by using `try`/`catch` and handling `ENOENT` + retry..